### ISSUE TITLE
expires property should apply to Product

### ISSUE THREAD

**Issue by MatthiasWiesmann** [schemaorg/schemaorg#3421]
The `CreativeWork` object has an `expires` property defined as follow:

> Date the content expires and is no longer useful or available.

Many products have an expiration date, which is basically the same. I would suggest the domain of this property be extended to also apply to Product.

**Comment by mfhepp** [comment]
Valid point. A few remarks, however:
1. We would need to adjust the text for schema:expires, as
"Date the content expires and is no longer useful or available. For example a [VideoObject](https://schema.org/VideoObject) or [NewsArticle](https://schema.org/NewsArticle) whose availability or relevance is time-limited, or a [ClaimReview](https://schema.org/ClaimReview) fact check whose publisher wants to indicate that it may no longer be relevant (or helpful to highlight) after some date."
may not fit. 
The current property mixes the aspects of **(1) availability** with **(2) end of relevance/usefulness/fitness for consumption**.
`expires` for media seems to be mostly useful for deciding whether an object should be shown or no longer, whereas for products, it will mostly be used to predict shelf life. It can be a useful feature, e.g. for comparing long-lasting variants of food products,
2. If you want to use a date or datetime as the range, you can only use the property on [IndividualProduct](https://schema.org/IndividualProduct), because every bottle of milk will have an individual expiry date. So we would have to include [Duration](https://schema.org/Duration) to the range. **This would change the meaning and may require changing consuming applications,** as a duration is **relative to some (often unknown) point in time** (e.g. date of production or purchase).

So I think **it would be better to add a new property or two new properties,** e.g. `expirationDate` with a range of date or datetime (for "best before use", "use by" data of individual products, like food items or pharmaceuticals, and `shelfLife`(we may need a better term) with a range of duration as a typical product characteristics.

A good reference is the [NSF page on expiration dates](https://www.nsf.org/consumer-resources/articles/expiration-dates).

**Comment by MatthiasWiesmann** [comment]
That makes sense, a new property on IndividualProduct would make sense.

**Comment by mfhepp** [comment]
> That makes sense, a new property on IndividualProduct would make sense.

I think the `shelfLife` property is more useful, because at least in public Web content, we rarely describe individual products or production batches (this may be different for B2B usages of the schema.org datamodel).

A bit more background is in the archived [GoodRelations Wiki](http://wiki.goodrelations-vocabulary.org/Documentation/Product_or_Service).

**Comment by MatthiasWiesmann** [comment]
The _period after opening_ would also be a candidate, it is very often present on cosmetic symbols, in a coded way: https://wiesmann.codiferes.net/wordpress/archives/29169

**Comment by mfhepp** [comment]
Maybe it is best to align with GS1 properties in here? 
E.g. https://www.gs1.org/gs1-web-vocabulary

**Comment by MatthiasWiesmann** [comment]
Yes, maybe what we need is to mention these at the right (product) level

**Comment by MatthiasWiesmann** [comment]
The shelf life can be expressed using the uncefact vocabulary.

```
{
  "@context": {
    "unece": "https://vocabulary.uncefact.org/",
    "schema": "http://schema.org/"
  },
  "@type": [
    "schema:Product",
    "unece:TradeProduct"
  ],
  "unece:fromDeliveryLifeSpanMeasure": {
    "@type": "unece:SupplyChainPackaging",
    "unece:DurationUnitMeasureCode": "unece:DurationUnitMeasureCode#MON",
    "unece:DurationUnitMeasureTypeValue": 6
  },
  "schema:name": "Golden Ramen",
  "schema:image": "GoldenRamen.jpg",
  "schema:offers": {
    "@type": "schema:Offer",
    "availability": "https://schema.org/InStock",
    "price": "2.00",
    "priceCurrency": "USD"
  }
}
```

**Comment by MatthiasWiesmann** [comment]
Expiration date can be expressed using gs1 properties:

```
{
  "@context": {
    "gs1": "https://ref.gs1.org/voc/",
    "schema": "http://schema.org/"
  },
  "@type": [
    "schema:Product",
    "gs1:Product"
  ],
  "schema:name": "Château Snotty 2020",
  "gs1:bestBeforeDate": "2025-12-31",
  "schema:offers": {
    "@type": "schema:Offer",
    "availability": "https://schema.org/InStock",
    "price": "2.00",
    "priceCurrency": "USD"
  }
}
```
